Add batch check command and safety features to CLI

tomjn opened the pull request:
Main goal here is to add features so that when using vagrant we don't need to trigger a `sudo` call if we don't have to. That means checks and batch operations. Likewise accounting for if the user walks away and we prompt them for a sudo password, then other programs change the hosts file while we're waiting for permission to make changes

**This depends on https://github.com/goodhosts/hostsfile/pull/57**

Features added:
- Extended check command to support batch checking: `check <IP> <host1> <host2>...`
- Added --quiet flag for check command (only return exit codes)
- Added --any flag for check command (match ANY instead of ALL hosts)
- Added --backup global flag to create backup before modifying hosts
- Added --safe global flag to detect concurrent modifications
- Added flushHostsfile() helper that implements backup and safety checks

Command updates:
- add, remove, clean now use flushHostsfile() for safe writes
- Backward compatible: single-arg check behavior preserved

Dependencies:
- Updated go.mod to use local hostsfile library with improvements

Exit codes:
- 0: Success (host(s) found or operation completed)
- 1: Check failed (host not found)
- 2: Invalid arguments

tomjn committed:
Add batch check command and safety features to CLI

Features added:
- Extended check command to support batch checking: `check <IP> <host1> <host2>...`
- Added --quiet flag for check command (only return exit codes)
- Added --any flag for check command (match ANY instead of ALL hosts)
- Added --backup global flag to create backup before modifying hosts
- Added --safe global flag to detect concurrent modifications
- Added flushHostsfile() helper that implements backup and safety checks

Command updates:
- add, remove, clean now use flushHostsfile() for safe writes
- Backward compatible: single-arg check behavior preserved

Dependencies:
- Updated go.mod to use local hostsfile library with VVV improvements

Exit codes:
- 0: Success (host(s) found or operation completed)
- 1: Check failed (host not found)
- 2: Invalid arguments

This enables VVV vagrant plugin to:
- Check hosts without sudo (read-only operation)
- Only request sudo when hosts are actually missing

Mte90 commented:
To me make sense, what do you think @luthermonson?

luthermonson commented:
great feature, huge fan of this. can you fix the lint error and we'll get this merged

tomjn commented:
@luthermonson this is blocked by https://github.com/goodhosts/hostsfile/pull/57, once that's merged and released this PR can be updated to use that which will fix CI

luthermonson commented:
https://github.com/goodhosts/hostsfile/releases/tag/v0.1.7 released

tomjn committed:
Update hostsfile dependency to v0.1.7

- Update from v0.1.6 to v0.1.7 to use the newly released version
- Remove local replace directive as hostsfile improvements are now published
- This allows CI tests to pass by using the published module from GitHub

tomjn commented:
@luthermonson CI is passing now!

luthermonson commented:
nice ;)

tomjn commented:
@luthermonson @Mte90 any opposition to me merging?
